packages/phone-number-privacy/signer/src/signer/pnp/quota.service.ts

```diff
@@ -0,0 +1,385 @@
+import { NULL_ADDRESS } from '@celo/base'
+import { StableToken } from '@celo/contractkit'
+import {
+  ErrorMessage,
+  isVerified,
+  PnpQuotaRequest,
+  SignMessageRequest,
+} from '@celo/phone-number-privacy-common'
+import Logger from 'bunyan'
+import { Transaction } from 'knex'
+import { Counters, Histograms } from '../../common/metrics'
+import config from '../../config'
+import { getPerformedQueryCount, incrementQueryCount } from '../../database/wrappers/account'
+import { storeRequest } from '../../database/wrappers/request'
+import {
+  getBlockNumber,
+  getCeloBalance,
+  getContractKit,
+  getStableTokenBalance,
+  getTransactionCount,
+  getWalletAddress,
+  meter,
+} from '../../web3/contracts'
+import { IQuotaService, OdisQuotaStatusResult } from '../quota.interface'
+import { PnpSession } from './session'
+export interface PnpQuotaStatus {
+  queryCount: number
+  totalQuota: number
+  blockNumber: number
+}
+
+export class PnpQuotaService implements IQuotaService<SignMessageRequest | PnpQuotaRequest> {
+  public async checkAndUpdateQuotaStatus(
+    state: PnpQuotaStatus,
+    session: PnpSession<SignMessageRequest>,
+    trx: Transaction
+  ): Promise<OdisQuotaStatusResult<SignMessageRequest>> {
+    Histograms.userRemainingQuotaAtRequest.observe(state.totalQuota - state.queryCount)
+    let sufficient = state.queryCount >= state.totalQuota
+    if (sufficient) {
```

Comment:
It's not totally clear to me whether this variable is misnamed or if this conditional is flipped, but the name does not seem to match the behavior.

Reply:
oh my